test/e2e/helpers_test.go

```diff
@@ -500,3 +467,47 @@ func getEnvValueInContainer(container *corev1.Container, name string) string {
 	}
 	return value
 }
+
+// getTemporaryKubeconfig dumps an environment's kubeconfig to a temporary file.
+func getTemporaryKubeconfig(t *testing.T, env environments.Environment) (path string, cleanup func()) {
+	t.Log("creating a tempfile for kubeconfig")
+
+	kubeconfig, err := generators.NewKubeConfigForRestConfig(env.Name(), env.Cluster().Config())
+	require.NoError(t, err)
+	kubeconfigFile, err := os.CreateTemp(os.TempDir(), "manifest-tests-kubeconfig-")
+	require.NoError(t, err)
+	defer kubeconfigFile.Close()
+
+	cleanup = func() {
+		_ = os.Remove(kubeconfigFile.Name())
+	}
```

Comment:
Why not use `t.Cleanup()`? It will automatically be called when the test finishes

Reply:
Good point, didn't think about it! https://github.com/Kong/kubernetes-ingress-controller/pull/3378/commits/a7025f32613c1f4b35113b6a5502751bc2d85380